chemclipse/plugins/org.eclipse.chemclipse.chromatogram.xxd.peak.detector.supplier.firstderivative/src/org/eclipse/chemclipse/chromatogram/xxd/peak/detector/supplier/firstderivative/settings/PeakDetectorSettingsMSD.java

```diff
@@ -35,6 +43,22 @@
 	@JsonPropertyDescription(value = "Window Size: 3, 5, 7, ..., 45")
 	@EnumSelectionSettingProperty
 	private WindowSize windowSize = WindowSize.WIDTH_5;
+	@JsonProperty(value = "Ion Filter Mode", defaultValue = "EXCLUDE")
+	@EnumSelectionRadioButtonsSettingProperty
+	FilterMode filterMode = FilterMode.EXCLUDE;
+	@JsonProperty(value = "Filter Ions", defaultValue = "")
+	@StringSettingsProperty(regExp = "")
```

Comment:
an empty regexp is useless, see above, if no validation is desired, simply do not specify the property. Also see above for the Type of the property. To support IMarkedIons as a whole we might need to extend the UI so it provides the user with a real nice UI.

Reply:
If the desired value is not specified, it defaults to an unwanted regex. The API doc states, that, in case no regex is desired, an empty String should be provided. This did not work as expected, see https://github.com/eclipse/chemclipse/issues/36.
Concerning the input widget, see https://github.com/eclipse/chemclipse/issues/41.

Reply:
You can either specify multiline or regexp or both, if you does not specify any of the two simply ommit the annotation.

Reply:
Understood. See https://github.com/eclipse/chemclipse/issues/43.